aten/src/ATen/native/quantized/cpu/qembeddingbag.cpp

```diff
@@ -237,9 +236,13 @@ at::Tensor embedding_bag_byte_impl(
     offsets_include_last_val[M] = indices.numel();
     offsets_data = offsets_include_last_val.data();
   }
-
-  std::vector<int64_t> shape = {output_size, D};
-  auto output = at::empty(shape, weight.options().dtype(at::kFloat));
+  if (indices.dim() == 2 && is_embedding_op) {
+    std::vector<int64_t> shape = {indices.size(0), indices.size(1), D};
+    output = at::empty(shape, weight.options().dtype(at::kFloat));
```

Comment:
nit: we can move output to outside of the branch and declare `std::vector<int64_t> shape` outside of the branch instead.